controllers/ocsinitialization/ocsinitialization_controller.go

```diff
@@ -391,22 +393,39 @@ func (r *OCSInitializationReconciler) ensureOcsOperatorConfigExists(initialData
 
 func (r *OCSInitializationReconciler) getEnableTopologyKeyValue() string {
 
-	// return true even if one of the storagecluster has enabled it
 	for _, sc := range r.clusters.GetStorageClusters() {
-		if sc.Spec.ManagedResources.CephNonResilientPools.Enable {
+		if !sc.Spec.ExternalStorage.Enable && sc.Spec.ManagedResources.CephNonResilientPools.Enable {
+			// In internal mode return true even if one of the storageCluster has enabled it via the CR
 			return "true"
+		} else if sc.Spec.ExternalStorage.Enable {
+			// In external mode, check if the non-resilient storageClass exists
+			scName := util.GenerateNameForNonResilientCephBlockPoolSC(&sc)
+			storageClass := util.GetStorageClassWithName(r.ctx, r.Client, scName)
+			if storageClass != nil {
+				return "true"
+			}
 		}
 	}
 
 	return "false"
 }
 
+// In case of multiple storageClusters when replica-1 is enabled for both an internal and an external cluster, different failure domain keys can lead to complications.
+// To prevent this, when gathering information for the external cluster, ensure that the failure domain is specified to match that of the internal cluster (sc.Status.FailureDomain).
 func (r *OCSInitializationReconciler) getTopologyDomainLabelsKeyValue() string {
 
-	// return value from the internal storagecluster as failureDomain is set only in internal cluster
 	for _, sc := range r.clusters.GetStorageClusters() {
-		if !sc.Spec.ExternalStorage.Enable && sc.Status.FailureDomainKey != "" {
+		if !sc.Spec.ExternalStorage.Enable && sc.Spec.ManagedResources.CephNonResilientPools.Enable {
```

Comment:
we arent setting this filled for external cluster storagecluster, so this will be always false

Reply:
The case you are referring to is of an internal cluster, the external cluster case is in the next else if, also we have added a note.